meta-ostro/classes/ostro-image.bbclass

```diff
@@ -10,121 +10,245 @@ IMAGE_INSTALL = " \
+#    Base image plus login via getty and ssh, plus connectivity.
+#    This is what developers  are expected to start with when
+#    building their first image.
+# ostro-image-swupd-dev -
+#    Image used for testing Ostro OS. Contains most of the software
+#    pre-installed, including the corresponding development files
+#    for on-target compilation.
+# ostro-image-swupd-all -
+#    Contains all defined bundles. Useful as meta target, but not
+#    guaranteed to build images successfully, for example because
+#    the content might get too large for machines with a fixed image
+#    size.
+SWUPD_IMAGES ??= " \
+    reference \
+    dev \
+    all \
+"
+# SWUPD_IMAGES[reference] = " \
+#     connectivity \
+#     ssh-server \
+# "
+SWUPD_IMAGES[reference] = " \
+    reference \
+"
+
+# In practice the same as "all" at the moment, but conceptually different
+# and thus defined separately.
+SWUPD_IMAGES[dev] = " \
+    ${SWUPD_BUNDLES} \
 "
+SWUPD_IMAGES[all] = " \
+    ${SWUPD_BUNDLES} \
+"
+
+# When building without swupd, choose which content is to be included
+# in the image. If the default "ostro-image-noswupd" name is
+# undesirable, write a custom image recipe or customize the image file
+# names. We provide variables that can be used to select the same
+# content as in the swupd images.
+#
+# Example for local.conf, partly covered already by ostro-development.inc:
+# IMAGE_BASENAME_pn-ostro-image-noswupd = "my-ostro-image-reference"
+# OSTRO_IMAGE_EXTRA_INSTALL = "${OSTRO_IMAGE_INSTALL_REFERENCE} my-own-package"
+# OSTRO_IMAGE_EXTRA_FEATURES = "${OSTRO_IMAGE_FEATURES_REFERENCE}"
 
-# Once officially supported, variations with IMA disabled can be
-# added. Right now, users need to do that in their local.conf:
-# OSTRO_EXTRA_IMAGE_VARIANTS = "imagevariant:noima imagevariant:dev,noima"
+# Customize priorities of alternative components. See ostro.conf.
+#
+# In general, Busybox or Toybox are preferred over alternatives.
+# The expectation is that either Busybox or Toybox are used, but if
+# both get installed, Toybox is used for those commands that it
+# provides.
+#
+# It is still possible to build images with coreutils providing
+# core system tools, one just has to remove Toybox/Busybox from
+# the image.
+export ALTERNATIVE_PRIORITY_BUSYBOX ?= "300"
+export ALTERNATIVE_PRIORITY_TOYBOX ?= "301"
+export ALTERNATIVE_PRIORITY_BASH ?= "305"
+
+# Both systemd and the efi_combo_updater have problems when
+# "mount" is provided by busybox: systemd fails to remount
+# the rootfs read/write and the updater segfaults because
+# it does not parse the output correctly.
+#
+# For now avoid these problems by sticking to the traditional
+# mount utilities from util-linux.
+export ALTERNATIVE_PRIORITY_UTIL_LINUX ?= "305"
+
+# We do not know exactly which util-linux packages will get
+# pulled into bundles, so we have to install all of them
+# also in the os-core. Alternatively we could try to select
+# just mount/umount as overrides for Toybox/Busybox.
```

Comment:
nice catch, but if we can patch toybox to provide a more complete and compliant `mount`,  this can be avoided. Ask the team that does toybox, maybe they can solve this in parallel.